Expose cpe_dict_model_add_item()

Hexadorsimal opened the pull request:
Changed cpe_dict_model_add_item() to be a publicly accessible function.

openscap-jenkins commented:
Can one of the admins verify this patch?

rsprudencio commented:
@openscap-jenkins add to whitelist

rsprudencio commented:
@openscap-jenkins test this please

rsprudencio commented on src/CPE/cpedict_priv.c:
```diff
@@ -155,8 +155,7 @@ OSCAP_ACCESSOR_STRING(cpe_generator, product_name)
 OSCAP_ACCESSOR_SIMPLE(int, cpe_dict_model, base_version)
-OSCAP_IGETTER_GEN(cpe_item, cpe_dict_model, items)
-OSCAP_ITERATOR_REMOVE_F(cpe_item)
+OSCAP_IGETINS_GEN(cpe_item, cpe_dict_model, items, item) OSCAP_ITERATOR_REMOVE_F(cpe_item)
```
Any specific reason for this change?

Hexadorsimal commented:
The function cpe_dict_model_add_item() was static/private to that file it
was in, but I wanted to be able to add an item to a cpe_dict_model, so I
deleted the static function and changed the macro from OSCAP_IGETTER_GEN to
OSCAP_IGETINS_GEN, which will create the function for us publicly, so that
we can add cpe_items to cpe_dict_models.

On Fri, May 5, 2017 at 7:07 AM, Raphael Sanchez Prudencio <
notifications@github.com> wrote:

> *@rsprudencio* commented on this pull request.
> ------------------------------
>
> In src/CPE/cpedict_priv.c
> <https://github.com/OpenSCAP/openscap/pull/732#discussion_r114973675>:
>
> > @@ -155,8 +155,7 @@ OSCAP_ACCESSOR_STRING(cpe_generator, product_name)
>
>  OSCAP_GETTER(struct cpe_generator *, cpe_dict_model, generator)
>  OSCAP_ACCESSOR_SIMPLE(int, cpe_dict_model, base_version)
> -OSCAP_IGETTER_GEN(cpe_item, cpe_dict_model, items)
> -OSCAP_ITERATOR_REMOVE_F(cpe_item)
> +OSCAP_IGETINS_GEN(cpe_item, cpe_dict_model, items, item) OSCAP_ITERATOR_REMOVE_F(cpe_item)
>
> Any specific reason for this change?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/OpenSCAP/openscap/pull/732#pullrequestreview-36481384>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AL2B9JcuVKoaN6nSrxmstOxM2KeiOm6fks5r2wL4gaJpZM4NQv-O>
> .
>

mpreisler commented on src/CPE/cpedict_priv.c:
```diff
@@ -291,25 +288,9 @@ static void cpe_notes_export(const struct cpe_notes *notes, xmlTextWriterPtr wri
-}
 
 /***************************************************************************/
+
```
?

Hexadorsimal replied:
I didn't really do it on purpose, but it was how the code ended up after I deleted the function.  I was just balancing the blank lines above and below because I thought it looked nice.  We can change it if it bothers you.

mpreisler commented on src/CPE/public/cpe_dict.h:
```diff
@@ -522,6 +522,9 @@ bool cpe_item_add_title(struct cpe_item *item, struct oscap_text *new_title);
 OSCAP_DEPRECATED(bool cpe_item_add_note(struct cpe_item *item, struct oscap_text *new_title));
 
 /// @memberof cpe_dict_model
+bool cpe_dict_model_add_item(struct cpe_dict_model *dict, struct cpe_item *new_item);
```
Where is this function implemented?

Hexadorsimal replied:
The difference between OSCAP_IGETTER_GEN and OSCAP_IGETINS_GEN is that IGETINS also implements an insertion function (I'm assuming the INS stands for insert).  cpe_dict_model_add_item() is that insertion function.  It's just buried deep in a macro, but the prototype in the public header doesn't use the macro so it's easier for to read.  I didn't come up with this, I'm just following the convention that all the other functions were using in the module.

Hexadorsimal replied:
https://github.com/OpenSCAP/openscap/blob/maint-1.2/src/common/util.h#L291-L300

mpreisler replied:
Alright

mpreisler commented:
ACK. Thanks!
